Lombiq.Tests.UI/Extensions/OrchardFeaturesTestingUITestContextExtensions.cs

```diff
@@ -0,0 +1,199 @@
+                context.Exists(
+                    By.XPath("//div[contains(@class, 'alert-info') and contains(.,'This folder is empty')]"));
+
+                context.UploadSamplePngByIdOfAnyVisibility("fileupload"); // #spell-check-ignore-line
+                context.UploadSamplePdfByIdOfAnyVisibility("fileupload"); // #spell-check-ignore-line
+                context.WaitForPageLoad();
+
+                var image = context.Get(By.XPath($"//span[contains(text(), '{imageName}')]"));
+
+                context.Exists(By.XPath($"//span[contains(text(), '{documentName}')]"));
+
+                await image.ClickReliablyAsync(context);
+
+                await context
+                    .Get(By.XPath($"//span[contains(text(), '{imageName}')]/ancestor::tr"))
+                    .Get(By.CssSelector("a.btn.btn-link.btn-sm.delete-button"))
+                    .ClickReliablyAsync(context);
+
+                await context.ClickModalOkAsync();
+                context.WaitForPageLoad();
+                await context.GoToAdminRelativeUrlAsync(mediaPath);
+
+                context.Missing(By.XPath("//span[text()=' Image.png ' and @class='break-word']"));
+
+                var deleteFolderButton =
+                    context.Get(By.CssSelector("#folder-tree  li.selected  div.btn-group.folder-actions .svg-inline--fa.fa-trash"));
+                await deleteFolderButton.ClickReliablyAsync(context);
+
+                await context.ClickModalOkAsync();
+                context.WaitForPageLoad();
+                await context.GoToAdminRelativeUrlAsync(mediaPath);
+
+                context.Missing(By.XPath("//div[text()='Example Folder' and @class='folder-name ms-2']"));
+            });
+
+    public static Task TestAuditTrailAsync(this UITestContext context) =>
+    context.ExecuteTestAsync(
+        "Test Audit Trail",
+        async () =>
+        {
+            var auditTrailPath = "/AuditTrail";
+
+            await context.EnableFeatureDirectlyAsync("OrchardCore.AuditTrail");
+            await context.GoToAdminRelativeUrlAsync("/Settings" + auditTrailPath);
+
+            await context.GoToEditorTabAsync("Content");
+
+            await context.SetCheckboxValueAsync(By.XPath("//input[@value='Page']"), isChecked: true);
+
+            await context.ClickReliablyOnSubmitAsync();
+
+            var contentItemsPage = await context.GoToContentItemsPageAsync();
+            context.RefreshCurrentAtataContext();
+            contentItemsPage
+                .CreateNewPage()
+                    .Title.Set("Audit Trail Test Page")
+                    .Publish.ClickAndGo()
+                .AlertMessages.Should.Contain(message => message.IsSuccess);
+
+            await context.GoToAdminRelativeUrlAsync(auditTrailPath);
+
+            context.Exists(ByHelper.TextContains("of the Page"));
+
+            context.Exists(ByHelper.TextContains("was published"));
+
+            context.Exists(ByHelper.TextContains("was created"));
```

Comment:
These aren't unique to Audit Trail Test Page. Also check for "Audit Trail Test Page" to be included with these.